lib/xcodeproj/project/project_helper.rb

```diff
@@ -157,6 +157,15 @@ def self.configuration_list(project, platform, deployment_target = nil, target_p
 
         cl.build_configurations << release_conf
         cl.build_configurations << debug_conf
+
+        project.build_configurations.each do |configuration|
+          next if %w(Debug Release).include?(configuration.name)
```

Comment:
Also seems like the presence and contents of the default configurations isn't well-tested enough to really touch :)